pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -854,6 +854,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 	for _, egressListener := range node.SidecarScope.EgressListeners {
 
 		services := egressListener.Services()
+
+		// There is a possibility that egress listener might not have selected any service - early return from here.
+		if len(services) == 0 {
```

Comment:
@howardjohn @hzxuzhonghu Do you think it is useful - Looks like the whole logic below is only useful if egress listener has services?

Reply:
seems fine but I'm not sure what the motivation is? It doesnt look like we save on expensive work

Reply:
As I mentioned in the PR description, just a minor thing - NBD. Definitely we do not save on expensive work - except for few minor checks. If you think, it is not worth, happy to close this.

Reply:
May be not worth it - I will close this